### PR TITLE
Fix typos and invalid code fences languages

### PR DESCRIPTION
**Pull request by djhi**

### CODE REVIEW

File: docs/ListTutorial.md [hunk 1]
```diff
@@ -457,13 +464,20 @@ import {
...
     <List filters={filters}>
         <Datagrid>
-            <TextField source="id" />
-            <TextField source="title" />
-            <TextField source="author" />
-            <TextField source="year" />
+            <TextField<Book> source="id" />
+            <TextField<Book> source="title" />
+            <TextField<Book> source="author" />
+            <TextField<Book> source="year" />
```
Comment (fzaninotto) [inline docs/ListTutorial.md:480]: This is overkill IMHO

File: docs/ListTutorial.md [hunk 2]
```diff
@@ -539,12 +559,12 @@ export const PostList = () => {
                 />
             ) : (
                 <Datagrid rowClick="edit">
-                    <TextField source="id" />
-                    <ReferenceField label="User" source="userId" reference="users">
-                        <TextField source="name" />
+                    <TextField<Post> source="id" />
```
Comment (fzaninotto) [inline docs/ListTutorial.md:562]: this, too

**Review by fzaninotto — CHANGES_REQUESTED** [review]

File: docs/ListTutorial.md (hunk 1 above)
Comment (djhi) [inline docs/ListTutorial.md:480]:
```suggestion
            <TextField source="id" />
            <TextField source="title" />
            <TextField source="author" />
            <TextField source="year" />
```

File: docs/ListTutorial.md [hunk 3]
```diff
@@ -540,11 +560,11 @@ export const PostList = () => {
             ) : (
                 <Datagrid rowClick="edit">
                     <TextField source="id" />
-                    <ReferenceField label="User" source="userId" reference="users">
-                        <TextField source="name" />
+                    <ReferenceField<Post> label="User" source="userId" reference="users">
+                        <TextField<User> source="name" />
```
Comment (djhi) [inline docs/ListTutorial.md:564]:
```suggestion
                    <ReferenceField label="User" source="userId" reference="users">
                        <TextField source="name" />
```

File: docs/ListTutorial.md [hunk 4]
```diff
@@ -540,11 +560,11 @@ export const PostList = () => {
...
                     </ReferenceField>
-                    <TextField source="title" />
-                    <TextField source="body" />
+                    <TextField<Post> source="title" />
+                    <TextField<Post> source="body" />
```
Comment (djhi) [inline docs/ListTutorial.md:567]:
```suggestion
                    <TextField source="title" />
                    <TextField source="body" />
```

File: docs/ListTutorial.md [hunk 5]
```diff
@@ -522,11 +536,17 @@ import { List, SimpleList, Datagrid, TextField, ReferenceField } from 'react-adm
...
+    userId: number;
     title: string;
     views: number;
     published_at: string;
 }
 
+type User = {
+    id: number;
+    name: string;
+}
```
Comment (djhi) [inline docs/ListTutorial.md:548]:
```suggestion
```

File: docs/ListTutorial.md [hunk 6]
```diff
@@ -457,6 +464,13 @@ import {
 
 const filters = [<TextInput label="Search" source="q" size="small" alwaysOn />];
 
+type Book = {
+    id: string;
+    title: string;
+    author: string;
+    year: number;
+};
```
Comment (fzaninotto) [inline docs/ListTutorial.md:472]: Now useless

File: docs/ListTutorial.md [hunk 7]
```diff
@@ -39,11 +39,18 @@ import {
     TableCell,
 } from '@mui/material';
 
+type Book = {
+    id: string;
+    title: string;
+    author: string;
+    year: number;
+};
+
```
Comment (djhi) [inline docs/ListTutorial.md:48]:
```suggestion
```

**Review by fzaninotto — APPROVED** [review]

**Pull request merged**